File: TheaterSuggestions/Java/ScreeningSeatingArrangementSuggestions/src/main/java/org/weaveit/seatingplacesuggestions/PricingCategory.java
package org.weaveit.seatingplacesuggestions;

public enum PricingCategory {
    FIRST(1),
    SECOND(2),
    THIRD(3);

    private final int value;

    PricingCategory(int value) {
        this.value = value;
    }

    public int getValue() {
        return value;
    }

    public static PricingCategory valueOf(int value) {
        for (var category : values()) {
            if (category.value == value) {
                return category;
            }
        }
        throw new IllegalArgumentException("No category for value: " + value);
    }
}

File: TheaterSuggestions/Kotlin/ScreeningSeatingArrangementSuggestions/src/main/kotlin/org/weaveit/seatingplacesuggestions/PricingCategory.kt
package org.weaveit.seatingplacesuggestions

enum class PricingCategory(val value: Int) {
    FIRST(1),
    SECOND(2),
    THIRD(3);

    companion object {
        fun fromValue(value: Int): PricingCategory {
            return entries.find { it.value == value }
                ?: throw IllegalArgumentException("No category for value: $value")
        }
    }
}
